src/modules/admin/components/Permissions/ColonyPermissionEditDialog.tsx

```diff
@@ -340,34 +307,25 @@ const ColonyPermissionEditDialog = ({
                 />
               </div>
               <InputLabel label={MSG.permissionsLabel} />
-              {availableRoles.map((role, id) => (
+              {availableRoles.map((role, idx) => (
                 <div key={role} className={styles.permissionChoiceContainer}>
-                  <Checkbox
-                    className={styles.permissionChoice}
-                    value={role}
-                    name="roles"
+                  <PermissionCheckbox
                     disabled={!checkIfCanBeSet(role)}
-                  >
-                    <span className={styles.permissionChoiceDescription}>
-                      <Heading
-                        text={MSG[`role${id}`]}
-                        appearance={{ size: 'small', margin: 'none' }}
-                      />
-                      <FormattedMessage {...MSG[`roleDescription${id}`]} />
-                    </span>
-                  </Checkbox>
+                    role={role}
+                    roleDescription={MSG[`roleDescription${idx}`]}
```

Comment:
How come move the message handling for the role name to `PermissionCheckbox` but not for the description?

Reply:
So the role names are actually in the `en/i18n.json` file, but `PermissionCheckbox` does indeed have the _logic_ to piece together the message id strings.

That said, perhaps it would make sense to move the description messages into `PermissionCheckbox` and use the role name to access each message... I think I'll do that